module1/homework/make_unique/include/make_unique.hpp

```diff
@@ -0,0 +1,19 @@
+#pragma once
+
+#include <memory>
+#include <type_traits>
+
+namespace cs {
+template <typename T,
+          typename... Args,
+          typename = std::enable_if_t<!std::is_array_v<T>>>
+auto make_unique(Args&&... args) {
+    return std::unique_ptr<T>{new T(std::forward<Args>(args)...)};
+}
+
+template <typename T>
+auto make_unique(size_t size) {
+    return std::unique_ptr<T>{new std::remove_extent_t<T>[size] {}};
+}
```

Comment:
I'm curious why it doesn't leak. This should be a proper version for arrays.

```suggestion
template <typename T>
auto make_unique(size_t size) {
    return std::unique_ptr<T[]>{new std::remove_extent_t<T>[size] {}};
}
```

Reply:
T is array type as we passing it by template :)

Reply:
you are right, my mistake.